nav2_collision_monitor/src/collision_detector_node.cpp

```diff
@@ -308,6 +323,33 @@ void CollisionDetector::process()
     }
   }
 
+  if (collision_points_marker_pub_) {
```

Comment:
If you check if the publishers have subscriber > 1, then you can remove the parameter and just publish if anyone is listening. We didn't do this historically because that API used to not always work, but it does now. Also for things like the MPPI trajectories or costmaps, we do want to make sure you want them published at all since they're very heavy. This is not